Refactor #5139 - leftovers subscribe_to_all_hostgroups

dLobatog opened the pull request:
This functionality is replaced by simply creating a filter "view_hostgroups" , "unlimited" and assigning it to a role. This PR only cleans up the leftovers of it.

dLobatog committed:
Fixes #5139 - leftovers subscribe_to_all_hostgroups

Remove user_xxx unnecessary tables and notices

Update subhostgroups removed

Fixes for migration of foreign keys

Remove users from compute_resource fixture

Remove table notices after fk are removed for pg/mysql

domcleal commented on app/controllers/hostgroups_controller.rb:
```diff
@@ -44,7 +44,6 @@ def create
     if @hostgroup.save
       # Add the new hostgroup to the user's filters
-      @hostgroup.users << subscribed_users
       @hostgroup.users << users_in_ancestors
```
Should we remove all of user_hostgroup too?  Is there any use for it?

dLobatog replied:
I think they've been rendered useless now, @ares any idea if it's being used anywhere?

ares replied:
I don't think it's being used anywhere atm

ohadlevy replied:
There are a few of those, fact and or etc

dLobatog replied:
Right, would you mind if I get rid of them all in this PR?

domcleal replied:
Sure, let's have a cleanup.

domcleal commented:
Maybe the new auth system DB migration should add such a filter?

dLobatog commented:
[test] because Jenkins hiccup

isratrade commented:
@eLobato, need to rebase

dLobatog commented:
@isratrade rebased

isratrade commented:
looks good. not sure of the test failure, probably jenkins.

domcleal commented:
Failure of every single test is not likely to be Jenkins' fault.

isratrade commented:
I didn't test any test failures. It just says "Test Result (no tests)"

domcleal commented:
Click through all the red circles, then console output and it's failing to db:migrate.

dLobatog commented:
I've just rebased this again and fixed the errors that made `db:migrate` impossible, which were a typo and a few foreign keys that needed to be removed before dropping the user_xxx tables.

dLobatog commented:
Rebased again.

dLobatog commented:
Another rebase

dLobatog commented:
Latest fix fixes one test that got broken during rebase.

domcleal commented:
I'm happy with this, but @ohadlevy - could you confirm if there's anything you'll miss here?  (Notices are the only thing that stands out, but they're unused and we have planned to replace them one day.)

ohadlevy commented on app/views/api/v1/users/index.json.rabl:
```diff
@@ -1,5 +1,4 @@
 collection @users
 
 attributes :id, :login, :firstname, :lastname, :mail, :admin, :auth_source_id, :role_id, :last_login_on,
-           :domains_andor, :hostgroups_andor, :facts_andor, :filter_on_owner, :compute_resources_andor,
```
any chance this would break anything? should we consider returning empty values instead?

dLobatog replied:
After the permission system migration, these values are irrelevant, so it doesn't matter what they return, true/false, they were used to authorize users to domains, hostgroups, and so on, the new system just disregards this. 

If you're using api/v1 and Foreman > 1.5 it'll just mean you'll get default values for these fields, so clients can't really make any use of these. If someone is using these fields in their clients well it's making a big mistake as these fields do not matter at all in Foreman and are never updated according to the authorization system.

ohadlevy replied:
while i know they are irrelevant, i'm wondering if its going to break
anyone's automated scripts?

ohadlevy replied:
Also.. would you consider doing similar cleanup to javascript/css? ;)

dLobatog replied:
Sure, how about we keep api/v1 as is and mention in the docs that these values are not used anywhere and will always return empty strings to keep compatibility?

dLobatog replied:
Although I'm not quite sure where to put that information on the API docs...

domcleal replied:
You could add a "description" bit to the (API) UsersController#show action.

ohadlevy commented:
@domcleal @eLobato looks good, thanks for cleaning up!

dLobatog commented:
Latest update removes '/log' from the repository, `rails s` autocreates it.

dLobatog commented:
Rebased.

dLobatog commented:
I rebased this. I thought about adding a description to api/v1/users_controller#show, however removing these values (hostgroups_andor, domains_andor, etc...) from database will result in not returning them in the response even if defined in the RABL. I think that's OK.

dLobatog commented:
Rebased.

isratrade commented:
@eLobato, looks good. I suggest to add this commit https://github.com/isratrade/foreman/commit/66303b9bce50f84d31b7b66462f6c83de46413fa to preserve the response for API v1 even if the values are meaningless.

dLobatog commented:
@isratrade Thanks for your commit, I've put the changes in :smile:

dLobatog commented:
@isratrade , just rebased this again

isratrade commented:
@eLobato, failing test on DomainParameterTest

dLobatog commented:
@isratrade Thanks, I've updated it. :green_apple: now

isratrade commented:
Merged as 54141ab94976f2c398d94f314cf284fcb88fbc4a. thanks @eLobato !
